### PR TITLE
feat: Display cluster info on cluster details page

### PR DESCRIPTION
**Pull request by alexmt**
The PR is a continuation of https://github.com/argoproj/argo-cd/issues/2945 

Implemented changes add more information into the cluster secret and add cluster details page which presents collected data:

image

The page allows to view/edit cluster namespaces and name:

image

The administrator might invalidate cache using "INVALIDATE CACHE" button:

image

### CODE REVIEW

**Review by jannfis — APPROVED** [review]
LGTM with minor comments which don't necessarily need to be addressed.

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -183,5 +199,27 @@ func redact(clust *appv1.Cluster) *appv1.Cluster {
 	clust.Config.Password = ""
 	clust.Config.BearerToken = ""
 	clust.Config.TLSClientConfig.KeyData = nil
+	if clust.ConnectionState.ModifiedAt == nil || clust.ConnectionState.ModifiedAt.Add(clusterStateMaxAge).Before(time.Now()) {
```
[the remark this answers is not in the document]
Comment (alexmt) [reply server/cluster/cluster.go]: thank you ! done

File: pkg/apis/application/v1alpha1/types.go [hunk 2]
```diff
@@ -940,6 +940,19 @@ type Cluster struct {
...
+
+type ClusterCacheInfo struct {
+	// ResourcesCount holds number of observed Kubernetes resources
+	ResourcesCount int64 `json:"resourcesCount,omitempty" protobuf:"bytes,1,opt,name=resourcesCount"`
+	// APIsCount holds number of observed Kubernetes API count
+	APIsCount int64 `json:"apisCount,omitempty" protobuf:"bytes,2,opt,name=apisCount"`
+	// LastCacheSyncTime holds time of most recent cache synchronization
+	LastCacheSyncTime *metav1.Time `json:"lastCacheSyncTime,omitempty" protobuf:"bytes,3,opt,name=lastCacheSyncTime"`
+	// RefreshAt holds time when cluster cache refresh has been requested
+	RefreshAt *metav1.Time `json:"refreshAt,omitempty" protobuf:"bytes,4,opt,name=refreshAt"`
```
[the remark this answers is not in the document]
Comment (alexmt) [reply pkg/apis/application/v1alpha1/types.go]: Agree. Renamed

File: util/db/cluster.go [hunk 3]
```diff
@@ -54,7 +87,7 @@ func (db *db) listClusterSecrets() ([]*apiv1.Secret, error) {
 }
 
 // ListClusters returns list of clusters
-func (db *db) ListClusters(ctx context.Context) (*appv1.ClusterList, error) {
+func (db *db) ListClusters(_ context.Context) (*appv1.ClusterList, error) {
```
[the remark this answers is not in the document]
Comment (alexmt) [reply util/db/cluster.go:90]: reverted

File: util/db/cluster.go [hunk 4]
```diff
@@ -203,11 +223,11 @@ func (db *db) getClusterSecret(server string) (*apiv1.Secret, error) {
 }
 
 // GetCluster returns a cluster from a query
-func (db *db) GetCluster(ctx context.Context, server string) (*appv1.Cluster, error) {
+func (db *db) GetCluster(_ context.Context, server string) (*appv1.Cluster, error) {
```
[the remark this answers is not in the document]
Comment (alexmt) [reply util/db/cluster.go:226]: reverted

File: util/db/cluster.go [hunk 5]
```diff
@@ -71,13 +104,13 @@ func (db *db) ListClusters(ctx context.Context) (*appv1.ClusterList, error) {
...
 	if !hasInClusterCredentials {
-		clusterList.Items = append(clusterList.Items, localCluster)
+		clusterList.Items = append(clusterList.Items, *db.getLocalCluster())
 	}
 	return &clusterList, nil
 }
 
 // CreateCluster creates a cluster
-func (db *db) CreateCluster(ctx context.Context, c *appv1.Cluster) (*appv1.Cluster, error) {
+func (db *db) CreateCluster(_ context.Context, c *appv1.Cluster) (*appv1.Cluster, error) {
```
[the remark this answers is not in the document]
Comment (alexmt) [reply util/db/cluster.go:113]: reverted

**Pull request merged**